### PR TITLE
Use correct add instruction in inlineStringLatin1Inflate

### PR DESCRIPTION
**Pull request by dchopra001**
Fixes https://github.com/eclipse-openj9/openj9/issues/13647

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins test sanity zlinux jdknext

**Comment by dchopra001** [comment]
It's possible that this PR will also fix https://github.com/eclipse-openj9/openj9/issues/14124 and that's why so Peter has launched a sanity.functional build on javaNext in the comment above.

**Comment by dchopra001** [comment]
@joransiu Can I get a review please?

File: runtime/compiler/z/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -189,8 +189,8 @@ J9::Z::TreeEvaluator::inlineStringLatin1Inflate(TR::Node *node, TR::CodeGenerato
 
    // Done storing 16 chars. Now do some bookkeeping and then branch back to start label.
    generateRILInstruction(cg, TR::InstOpCode::AFI, node, srcOffRegister, 16);
-   generateRILInstruction(cg, TR::InstOpCode::AFI, node, sourceArrayReferenceRegister, 16);
-   generateRILInstruction(cg, TR::InstOpCode::AFI, node, charArrayReferenceRegister, 32);
+   generateRILInstruction(cg, cg->comp()->target().is64Bit() ? TR::InstOpCode::AGFI : TR::InstOpCode::AFI, node, sourceArrayReferenceRegister, 16);
+   generateRILInstruction(cg, cg->comp()->target().is64Bit() ? TR::InstOpCode::AGFI : TR::InstOpCode::AFI, node, charArrayReferenceRegister, 32);
```
Comment (joransiu) [inline runtime/compiler/z/codegen/J9TreeEvaluator.cpp:193]: Arguably, we could be using `LA sourceArrayReferenceRegister, 16/32(sourceArrayReferenceRegister)` here instead to compute the updated address.   That will handle both 31-bit and 64-bit patterns, and LA has a nicer bypass in the hardware when the target register is fed into memory references when consumed by the Load-Store Unit.

Comment (dchopra001) [reply runtime/compiler/z/codegen/J9TreeEvaluator.cpp]: Using LA now in 24cd8901ab3f18e45e45fb3cc5df0c6843b045b2

**Comment by joransiu** [comment]
> It's possible that this PR will also fix #14124 and that's why so Peter has launched a sanity.functional build on javaNext in the comment above.

This issue would only affect objects located near the end of a 4G boundary, where the existing 32-bit add instruction with +16 and +32 would fail to carry into the upper 32-bits.  I don't think this would address the object reference corruption we noticed in #14124 .

**Review by joransiu — APPROVED** [review]
LGTM.  Do you mind signing off your commit as per our [contributing guidelines](https://github.com/eclipse-openj9/openj9/blob/master/CONTRIBUTING.md#commit-guidelines)?

**Comment by joransiu** [comment]
jenkins test sanity zlinux

**Comment by dchopra001** [comment]
> LGTM. Do you mind signing off your commit as per our contributing guidelines?

Should be fixed now.

@mstoodle @0xdaryl This is a question for clarification. With eclipse removing the requirement to sign-off on commits earlier this year (https://www.eclipse.org/lists/eclipse.org-committers/msg01291.html), are OpenJ9 and OMR following suit? Or are commit sign-offs still mandatory?

**Comment by joransiu** [comment]
jenkins test sanity zlinux jdk11

**Comment by joransiu** [comment]
jenkins test sanity zlinux jdk17

**Pull request merged**